### ISSUE TITLE
Debian Stretch: jepsen.control.util/grepkill! fails if pattern does not match

### ISSUE THREAD

**Issue by GJL** [jepsen-io/jepsen#366]
On Debian Stretch, if the pattern passed to `grepkill!` does not match any processes, an exception is thrown:
```
Caused by: clojure.lang.ExceptionInfo: throw+: {:type :jepsen.control/nonzero-exit, :cmd "sudo -S -u root bash -c \"cd /; ps aux | grep doesnotexist | grep -v grep | awk \\\"{print \\\\\\$2}\\\" | xargs kill -9\"", :exit 123, :out "", :err "\nUsage:\n kill [options] <pid> [...]\n\nOptions:\n <pid> [...]            send signal to every <pid> listed\n -<signal>, -s, --signal <signal>\n                        specify the <signal> to be sent\n -l, --list=[<signal>]  list all signal names, or convert one to a name\n -L, --table            list all signal names in a nice table\n\n -h, --help     display this help and exit\n -V, --version  output version information and exit\n\nFor more details see kill(1).\n", :host "172.31.32.163", :action {:cmd "sudo -S -u root bash -c \"cd /; ps aux | grep doesnotexist | grep -v grep | awk \\\"{print \\\\\\$2}\\\" | xargs kill -9\"", :in "root\n"}}
```

I think the reason is that the `xargs` behavior changed in Stretch. The command below exits with code `123` when using Stretch (however, on Jessie, it returns with code `0`):

```
ps aux | grep doesnotexist | grep -v grep | awk "{print \$2}" | xargs kill -9
```

A possible solution could be to use xargs' `--no-run-if-empty` flag:

```
-r, --no-run-if-empty
              If  the  standard  input does not contain any nonblanks, do not run the command.  Normally, the command is run once even if there is no input.  This option is a
              GNU extension.
```

**Comment by aphyr** [comment]
Yeah, I think that'd be a good call!

**Comment by GJL** [comment]
Fixed via e778af5feddf7a679b6981ef71f6903b5e57320c
